### ISSUE TITLE
Pasteboard type

### ISSUE THREAD

**Issue by michalwski** [bitwarden/desktop#90]
I'd like to exclude Bitwarden from clipboard menu application like (Flycut). To do that I need to now what pasteboard type is used. According to this issue: https://github.com/bitwarden/browser/issues/326 it looks like this should be possible with the desktop version, is it?

**Comment by michalwski** [comment]
Flycut can be configured to exclude given pasteboard types. The default value it excludes is: `PasswordPboardType`. You can see it in preferences. Hope that helps.

**Comment by kspearrin** [comment]
Unfortunately I am not seeing an API in electron for defining the Pasteboard Type during clipboard write functions. https://electronjs.org/docs/api/clipboard#clipboardwritetexttext-type

There is a "type" parameter, but that doesn't seem related to the Pasteboard type, since all still seem to written as NSStringPboardType.

**Comment by goncalossilva** [comment]
They are marked as experimental APIs, but [this](https://www.electronjs.org/docs/api/clipboard#clipboardwritebufferformat-buffer-type-experimental) should work, where format is `org.nspasteboard.ConcealedType`.

**Comment by kspearrin** [comment]
Thanks @goncalossilva . We'll keep an eye on this API.

**Comment by 3zero2** [comment]
Have there been any updates about this by any chance?

**Comment by Stickyhavr** [comment]
> How do I view what the clipboard value's pasteboard type is?

One option is: https://github.com/sindresorhus/Pasteboard-Viewer

I found that info here: https://github.com/p0deje/Maccy/issues/125

I would love to be able to exclude BitWarden from Maccy. Hope this gets resolved someday. Thanks

**Comment by Stickyhavr** [comment]
> How do I view what the clipboard value's pasteboard type is?

One option is: https://github.com/sindresorhus/Pasteboard-Viewer

I found that info here: https://github.com/p0deje/Maccy/issues/125

I would love to be able to exclude BitWarden from Maccy. Hope this gets resolved someday. Thanks

**Comment by 3zero2** [comment]
Bitwarden only returns `public.utf8-plain-text` as the NSPasteboard type. I think that adding this to Flycut or Maccy would actually filter out not only Bitwarden but a bunch of other apps as well.

**Comment by Stickyhavr** [comment]
Yes, it would. That's why it would be best for BitWarden to use a particular pasteboard type that's not plain text. Either something like org.nspasteboard.ConcealedType or its own unique identifier. I don't know much about all of that, but there seems to be a list of some types here: http://nspasteboard.org/

**Comment by krabf** [comment]
Any updates on this? I'm also using Macccy and would love to exclude Bitwarden

**Comment by 3zero2** [comment]
No updates as far as I know.

**Comment by rraihansaputra** [comment]
Hi @kspearrin, apologies for the ping, but is there any update about the `clipboard.writeBuffer` API usage? The Electron docs still marks them as experimental, and the Bitwarden app still outputs `NSStringPboardType` (checked from Flycut).

**Comment by Stickyhavr** [comment]
I still have my fingers crossed this will show up someday.

**Comment by mcotse** [comment]
Any updates? Would love to keep using Bitwarden and have proper integration with various clipboard managers 🙏

**Comment by ChrisFR06** [comment]
_Any updates?_

**Comment by eliykat** [comment]
This has been triaged and is in our backlog, but does not have any specific deadline for a fix.
That said, last I looked into it, the [Electron](https://www.electronjs.org/docs/api/clipboard#clipboardwritebufferformat-buffer-type-experimental) and [Browser APIs](https://developer.mozilla.org/en-US/docs/Web/API/Clipboard_API) that let us set the pasteboard type are all marked as experimental. In particular, it looks like FF has limited support for this. I think there's more danger in implementing this in an inconsistent or unreliable fashion than waiting for the API support to catch up.

**Comment by bckp** [comment]
Maybe @kspearrin, @eliykat  this can be implemented in some: experimental feature in BW itself... so users need to enable that explicitly. That will help mac users that use Maccy (still do not undestand Apple do not implementet this directly) and if everythink goes wrong, you can just unckeck one option.

**Comment by ChrisFR06** [comment]
Has there been any new developments concerning this issue?

**Comment by spinlud** [comment]
+1. Shouldn't this be put in priority? It seems a concerning security risk for anyone using the etension and a clipboard manager 😮

**Comment by tomasherman** [comment]
for what it's worth - a workaround for Alfred users, you can set up alfred to ignore clipboard entries from certain apps - for example bitwarden :) This almost forced me to go back to 1password, luckily i found this workaround :)

![image](https://user-images.githubusercontent.com/99870/164974168-8773794b-5d97-4f0d-bc0b-851628310150.png)

**Comment by Stickyhavr** [comment]
Sure, for the desktop app it’s not a problem. Any good clipboard manager
should allow you to specify apps that it doesn’t copy. (Also any good
clipboard manager will automatically ignore a concealed type.)

The problem is with the browser extension because 99% of the things I copy
in my browser are things I want to be in the history. To make things even
messier, I actually like that my TOTP codes show up in my manager because I
have mine (Maccy) configured to show last copied item in the menu bar, so I
never have to wonder what’s on the clipboard, and I instantly know whether
a login has TOTP stored in Bitwarden, or if I need to go look for it
somewhere else, just by glancing at the menu bar.

So I’m still hoping this will come eventually. The most elegant solution
would be for Bitwarden to mark its copies as concealed, or its own unique
type. But thanks for posting a workaround, it’s nice to keep some energy in
this thread.

**Comment by tomasherman** [comment]
BTW this also affects credit card numbers, which to me is even worse than passwords altho of course passwords are bad enough. I wonder how/if other password managers address this in the browser extension.
